internal/cli/atlas/deployments/setup.go

```diff
@@ -0,0 +1,238 @@
+	}
+
+	networkName := fmt.Sprintf("mdb-local-%s", opts.deploymentName)
+	if _, err := podmanOpts.CreateNetwork(networkName); err != nil {
+		return err
+	}
+
+	// setup mongod
+	mongodDataVolume := fmt.Sprintf("mongod-local-data-%s", opts.deploymentName)
+	if _, err := podmanOpts.CreateVolume(mongodDataVolume); err != nil {
+		return err
+	}
+
+	if _, err := podmanOpts.RunContainer(
+		podman.RunContainerOpts{
+			Detach:   true,
+			Image:    fmt.Sprintf("mongodb/mongodb-enterprise-server:%s-ubi8", opts.mdbVersion),
+			Name:     mongodContainerName,
+			Hostname: InternalMongodHostname,
+			Volumes: map[string]string{
+				mongodDataVolume: "/data/db",
+			},
+			Ports: map[int]int{
+				opts.port: internalMongodPort,
+			},
+			Network: networkName,
+			Args: []string{
+				"--noauth",
+				"--dbpath", "/data/db",
+				"--replSet", ReplicaSetName,
+				"--setParameter", fmt.Sprintf("mongotHost=%s:%d", InternalMongotHostname, internalMongotPort),
+				"--setParameter", fmt.Sprintf("searchIndexManagementHostAndPort=%s:%d", InternalMongotHostname, internalMongotPort),
+				"--setParameter", "skipAuthenticationToMongot=true",
+			},
+		}); err != nil {
+		return err
+	}
+
+	// init ReplicaSet
+	if err := opts.waitConnection(opts.port); err != nil {
+		return err
+	}
+
+	seedRs := fmt.Sprintf(`try {
+		rs.status();
+	  } catch {
+		rs.initiate({
+		  _id: "%s",
+		  version: 1,
+		  configsvr: false,
+		  members: [{ _id: 0, host: "%s:%d", horizons: { external: "localhost:%d" } }],
+		});
+	  }`,
+		ReplicaSetName,
+		InternalMongodHostname,
+		internalMongodPort,
+		opts.port)
+	if err := opts.seed(opts.port, seedRs); err != nil {
+		return err
+	}
+
+	// setup mongot
+	mongotDataVolume := fmt.Sprintf("mongot-local-data-%s", opts.deploymentName)
+	if _, err := podmanOpts.CreateVolume(mongotDataVolume); err != nil {
+		return err
+	}
+
+	mongotMetricsVolume := fmt.Sprintf("mongot-local-metrics-%s", opts.deploymentName)
+	if _, err := podmanOpts.CreateVolume(mongotMetricsVolume); err != nil {
+		return err
+	}
+
+	_, err := podmanOpts.RunContainer(podman.RunContainerOpts{
+		Detach:   true,
+		Image:    "mongodb/apix_test:mongot-noauth",
+		Name:     fmt.Sprintf("%s-%s", InternalMongotHostname, opts.deploymentName),
+		Hostname: InternalMongotHostname,
+		Volumes: map[string]string{
+			mongotDataVolume:    "/var/lib/mongot",
+			mongotMetricsVolume: "/var/lib/mongot/metrics",
+		},
+		Network: networkName,
+	})
+
+	return err
+}
+
+func connString(port int) string {
+	return fmt.Sprintf("mongodb://localhost:%d/admin", port)
+}
+
+func (opts *SetupOpts) seed(port int, script string) error {
+	return mongosh.Exec(opts.debug, connString(port), "--eval", script)
+}
+
+func (opts *SetupOpts) waitConnection(port int) error {
+	const waitSeconds = 60
+	for i := 0; i < waitSeconds; i++ {
+		if err := mongosh.Exec(opts.debug, connString(port), "--eval", "db.runCommand('ping').ok"); err == nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
could this be done with the go driver?

Reply:
this requires testing but the first example in https://github.com/mongodb/mongo-go-driver#usage seems close

Reply:
run into some issues while testing and I created https://jira.mongodb.org/browse/CLOUDP-196391 in the same epic